packages/shared/src/components/post/write/WriteLinkPreview.tsx

```diff
@@ -15,34 +15,48 @@ import { ExternalLinkPreview } from '../../../graphql/posts';
 interface WriteLinkPreviewProps {
   link: string;
   preview: ExternalLinkPreview;
-  onLinkChange: FormEventHandler<HTMLInputElement>;
+  onLinkChange?: FormEventHandler<HTMLInputElement>;
   className?: string;
+  showPreviewLink?: boolean;
+  isMinimized?: boolean;
 }
 
 export function WriteLinkPreview({
   link,
   preview,
   onLinkChange,
   className,
+  isMinimized,
+  showPreviewLink = true,
 }: WriteLinkPreviewProps): ReactElement {
   return (
     <WritePreviewContainer className={className}>
-      <TextField
-        leftIcon={<LinkIcon />}
-        label="URL"
-        type="url"
-        name="url"
-        inputId="preview_url"
-        fieldType="tertiary"
-        className={{ container: 'w-full' }}
-        value={link}
-        onInput={onLinkChange}
-      />
+      {showPreviewLink && (
+        <TextField
+          leftIcon={<LinkIcon />}
+          label="URL"
+          type="url"
+          name="url"
+          inputId="preview_url"
+          fieldType="tertiary"
+          className={{ container: 'w-full' }}
+          value={link}
+          onInput={onLinkChange}
+        />
+      )}
       {preview.title && preview.image && (
-        <WritePreviewContent>
+        <WritePreviewContent className={isMinimized && '!py-2 !px-3'}>
           <div className="flex flex-col flex-1 typo-footnote">
             <span className="font-bold line-clamp-2">{preview.title}</span>
-            {preview.source && (
+            {preview.source && isMinimized ? (
+              <div className="flex absolute right-28 justify-center items-center w-8 h-8 rounded-full bg-theme-bg-primary">
```

Comment:
Is this wrapper needed here?
I thought the SourceAvatar handled most of this? (but could be wrong)

Reply:
You are right! Just needed a few classes but we should not need the wrapper.